### PR TITLE
Support Roche requirements and fixes

### PR DESCRIPTION
**Pull request by yanchengnv**
Fixes # .

This PR implements the following:
- Enhanced UnsafeJob processing by proactively aborting the job on all clients when UnsafeJob error is returned.
- Implemented event-based unsafe component detection and processing (see https://docs.google.com/document/d/1g07KTFBwcQS1vJjzz1icYCcZwWeTLmN3na1DauwPLYA/edit)
- Define gRPC's keepalive_time_ms to 120000 as default value. This is to make gRPC's keepalive more frequent to avoid the problem that the network provider reset TCP after idling too long.
- Improved cancelled task handling in AIO
- Improved logger creation by naming with qualified class name. This will make it easier to configure loggers to different log level.

A few sentences describing the changes proposed in this pull request.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Quick tests passed locally by running `./runtest.sh`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated.

### CODE REVIEW

File: nvflare/apis/fl_exception.py [hunk 1]
```diff
@@ -33,3 +33,9 @@ class UnsafeJobError(Exception):
     """Raised when a job is detected to be unsafe"""
 
     pass
+
+
+class UnsafeComponentError(Exception):
+    """Raised when configuration parsing error happens."""
+
+    pass
```
Comment (YuanTingHsieh) [inline nvflare/apis/fl_exception.py:41]: What about """Raised when a component in the configuration is detected to be unsafe"""?

Comment (yanchengnv) [reply nvflare/apis/fl_exception.py]: Changed as suggested.

**Review by SYangster — APPROVED** [review]
Passes test plan, LGTM

**Pull request merged**